python/cudf/cudf/core/dataframe.py

```diff
@@ -6520,17 +6521,43 @@ def _apply_support_method(self, method, axis=0, *args, **kwargs):
                 raise NotImplementedError(msg)
 
             prepared = self._prepare_for_rowwise_op(method)
+            if skipna is False and any(
+                [col.nullable for col in prepared._columns]
+            ):
+                mask = cudf.DataFrame(
+                    {
+                        name: cudf._lib.transform.mask_to_bools(
+                            prepared._data[name].base_mask,
+                            0,
+                            len(prepared._data[name]),
+                        )
+                        if prepared._data[name].nullable
+                        else column.full(len(prepared._data[name]), True)
+                        for name in prepared._data.names
+                    }
+                )
+                mask = mask.all(axis=1)
+            else:
+                mask = None
+
             prepared = prepared.astype("float64")
             prepared = prepared.fillna(np.nan)
 
             arr = cupy.asarray(prepared.as_gpu_matrix())
 
-            if method in _cupy_nan_methods_map:
+            if skipna is not False and method in _cupy_nan_methods_map:
                 method = _cupy_nan_methods_map[method]
 
             result = getattr(cupy, method)(arr, axis=1, **kwargs)
 
             if len(result.shape) == 1:
+                result = column.as_column(result)
+                if mask is not None:
+                    result = column.build_column(
+                        data=result.data,
+                        dtype=result.dtype,
+                        mask=cudf._lib.transform.bools_to_mask(mask._column),
+                    )
```

Comment:
Instead of going through `build_column` for this I believe we have a `set_mask` API that would be more ideal.